### ISSUE TITLE
Non-standard Tree selection behaviour

### ISSUE THREAD

**Issue by Heydon** [iTwin/stratakit#1601]
It has been noticed that tree item selection behavior is non-standard and should be changed to meet established conventions.

## Expected behavior

* Clicking a tree item selects the item and highlights it
* Clicking another tree item deselects the previously selected item and selects the new one (this is comparable to radio button/group behavior).
* Holding <kbd>Cmd</kbd> (or <kbd>Ctrl</kbd> on Windows) and clicking another item will mean two items are now selected.
* Holding <kbd>Cmd</kbd> (or <kbd>Ctrl</kbd>) and clicking will accumulate additional items.
* Holding <kbd>Shift</kbd> and clicking an item will keep the previously clicked item selected, select the newly targeted item, and select all items between these two.

In short, standard file system selection behaviour is expected and preferred.

## Current behavior

* Clicking an item selects the item. This is more akin to checkbox—not radio—behaviour.
* Clicking further items selects them additionally.
* Accumulated selected items cannot be removed with a single action. They have to be re-clicked individually.

<kbd>Shift</kbd> + click and <kbd>Cmd</kbd> (or <kbd>Ctrl</kbd>) + click are not implemented.

## Proposed solution

1. Basic selection behavior should be exclusive. When one item is selected, all others become unselected.
2. Using the <kbd>Shift</kbd> and <kbd>Cmd</kbd> (or <kbd>Ctrl</kbd>) modifier keys must be the only method for selecting multiple items.

**Comment by mayank99** [comment]
Is this about Toolbar or Tree? Neither of them have any built-in selection behavior (it's all configured on the consumer end).

Where are you seeing the current behavior? More details are necessary before we can take any action here.

**Comment by Heydon** [comment]
@mayank99 Tree!! Not sure how I managed that. Changed the title.

It seems the default behavior is to select items cumulatively, as described. That's not what many are looking for. 

I'm thinking if they need to implement <kbd>shift</kbd> and <kbd>cmd/ctrl</kbd> themselves, they would probably want this change in pointer selection behavior in place first, to build on top of.

**Comment by mayank99** [comment]
Clarified synchronously today: there is no "default" behavior. Single vs multiple selection are controlled by individually setting the [`selected`](https://stratakit.bentley.com/docs/reference/structures/Tree/#Tree.Item.selected) prop on each `Tree.Item`. Examples of this can be added when the [`Tree`](https://stratakit.bentley.com/docs/components/tree/) component is fully documented.

Note: <kbd>shift</kbd> and other modifiers should not be _required_ to select multiple items, as it excludes users on pointer-only devices. Cumulative pointer selection avoids this potential accessibility issue.
